pkg/composition/placement-request.go

```diff
@@ -0,0 +1,87 @@
+package composition
+
+import (
+	"fmt"
+	"slices"
+
+	"kombiner/pkg/apis/v1alpha1"
+)
+
+// PlacementRequest purpose is to provide auxiliar methods to an existing
+// v1alpha1.PlacementRequest.
+type PlacementRequest struct {
+	*v1alpha1.PlacementRequest
+}
+
+// SetBindingFailure helps when adding a bind failure condition to the
+// PlacementRequest.
+func (pr *PlacementRequest) SetBindingFailure(bind v1alpha1.Binding, reason, msg string) {
```

Comment:
Unless keeping a state outside a PR is required it's more suitable to use `SetBindingFailure(pr *PlacementRequest, bind v1alpha1.Binding, reason, msg string)` as the signature to avoid the wrapper.

The same argument for other cases.

Reply:
I dislike this approach so much I don't even know how to start. I haven't yet seen a convincing argument for having them. That being said, I will let you decide.